core/src/main/java/feast/core/model/Feature.java

```diff
@@ -32,19 +32,49 @@
 @Getter
 @Setter
 @Entity
-public class Feature extends Field {
+public class Feature {
+
+  @EmbeddedId private FeatureReference reference;
+
+  // Type of the field
+  private String type;
+
+  // Presence constraints (refer to proto feast.core.FeatureSet.FeatureSpec)
+  // Only one of them can be set.
+  private byte[] presence;
+  private byte[] groupPresence;
+
+  // Shape type (refer to proto feast.core.FeatureSet.FeatureSpec)
+  // Only one of them can be set.
+  private byte[] shape;
+  private byte[] valueCount;
+
+  // Domain info for the values (refer to proto feast.core.FeatureSet.FeatureSpec)
+  // Only one of them can be set.
+  private String domain;
+  private byte[] intDomain;
+  private byte[] floatDomain;
+  private byte[] stringDomain;
+  private byte[] boolDomain;
+  private byte[] structDomain;
+  private byte[] naturalLanguageDomain;
+  private byte[] imageDomain;
+  private byte[] midDomain;
+  private byte[] urlDomain;
+  private byte[] timeDomain;
+  private byte[] timeOfDayDomain;
 
   public Feature() {}
 
   private Feature(String name, ValueType.Enum type) {
-    this.setId(new FieldId());
-    this.setName(name);
+    this.setReference(new FeatureReference());
```

Comment:
Is there a reason to have a no args constructor for `FeatureReference`? If there isn't then I think it might be safer to require all the fields/args that make up the reference.

Reply:
Using a constructor with just name now. The previous model refactor you did injects the project and feature set name independently of the initialisation of the Feature/Entity class. I don't think it's within scope to change that right now.